teps/0040-ignore-step-errors.md

```diff
@@ -148,10 +158,135 @@ This proposal is limited to a step within a task and does not address `pipelineT
+
+    // desired exit code of the container
+    //exitCode takes only integer in the range 0 - 255
+    //when set, the entrypoint replaces the exit code of the wrapped command with the value specified here
+    ExitCode int `json:"exitCode,omitempty"`
+}
+```
+
+A `task` author or a `pipeline` author can use `exitCode` to specify their desired exit code for the step. If `exitCode` is
+set, the entrypoint replaces the original exit code of the `script` or the wrapped command with the value specified in
+the `exitCode`.
+
+To ignore a step error, set the `exitCode` to `0`:
+
+```yaml
+steps:
+  - image: docker.io/library/golang:latest
+    name: ignore-unit-test-failure
+    exitCode: 0
+    script: |
+      go test .
+```
+
+A `step` with `exitCode` set to `0` results in a successful execution and continues executing the rest of the steps in
+a task.
+
+The original exit code of the wrapped command is available in the status through the `taskResults`.
+The original exit code is stored as a task result with a key `step-<step-name>-exitcode` where `<step-name>` is the name
+of the step. If the step name is not defined, the task result key is set to `step-unnamed-<step-number>-exitcode`, where
+`<step-number>` is 0 for the first step, 1 for the second step, so on and so forth.
+
+In the following examples, the entrypoint in the first step `task1-failing-step` exited with `1` and later replaced the
+exit code with the one specified in the `exitCode`. Similarly, in the next `pipelineTask`, the step `task2-failing-step`
+exited with `1` and later replaced the exit code with the one specified in the `exitCode`:
+
+#### `pipelineRun` with a failing step
+
+```shell
+kubectl get pr  pipelinerun-with-failing-step -o json | jq .status.taskRuns | jq 'map(.status.taskResults)'
+[
+  [
+    {
+      "name": "step-task1-failing-step-exitcode",
+      "value": "1"
+    }
+  ],
+  [
+    {
+      "name": "step-task2-failing-step-exitcode",
+      "value": "1"
+    }
+  ]
+]
+```
+
+#### `taskRun` with a failing step
+
+```shell
+kubectl get tr taskrun-with-failing-step -o json | jq .status.taskResults
+[
+  {
+    "name": "step-failing-step-1-exitcode",
+    "value": "123"
+  },
+  {
+    "name": "step-failing-step-2-exitcode",
+    "value": "200"
+  }
+]
+```
+
+#### `taskRun` with a first failing step without any name
+
+```shell
+kubectl get tr taskrun-with-unamed-failing-step -o json | jq .status.taskResults
+[
+  {
+    "name": "step-unnamed-0-exitcode",
+    "value": "1"
+  }
+]
+```
+
+This new field `exitCode` will be implemented as a `alpha` feature and can be enabled by setting `enable-api-fields`
+to `alpha`.
+
+### Advantages
+
+#### exitCode set to 1 through 255
+
+  This provides an option to a `task` author or a `pipeline` author to overwrite the original exit code with their
+  desired value. At the same time, it provides an option to continue treating that step as a failure since the step
+  exiting with any non-zero is considered a failure.
```

Comment:
Do we want to support *additional* values ? I can really only think of one which would be "any exit code but 0" for example (aka I want to the process to fail in the step but I don't care with which code).

Reply:
> Do we want to support additional values ? 

we can if needed, how about `256` 🙃  i.e. `1` for "any exit code but 0", any other suggestions? 😄